cpp/include/cudf/io_types.h

```diff
@@ -54,7 +54,7 @@ typedef struct {
   bool          skipinitialspace;           /**< skip white spaces after the delimiter - default is false                                       */
 
 
-  int           nrows;                      // number of rows to read,  -1 indicates all
+  long          nrows;                      // number of rows to read,  -1 indicates all
```

Comment:
```suggestion
  gdf_size_type          nrows;                      // number of rows to read,  -1 indicates all
```

Reply:
@jrhemstad With the current implementation, nrows has to be a signed integer, because it needs to have an "invalid" value, with is -1 ATM. 
I can add a flag to denote if the nrows should be used, or pass nrows using a pointer. I'm not a big fan or either of the options. 
Please suggest the preferred course of action here.

Reply:
Hm, https://github.com/rapidsai/cudf/pull/583 will fix this problem as it changes `gdf_size_type` to `int`. However, your PR is higher priority. So I would say make `nrows` into `int` with a TODO to change to `gdf_size_type`.